Sources/TokamakLint/TokenVisitor.swift

```diff
@@ -0,0 +1,234 @@
+//
+//  TokenVisitor.swift
+//  SwiftSyntax
+//
+//  Created by Matvii Hodovaniuk on 4/2/19.
+//
+
+import Foundation
+import SwiftSyntax
+
+class TokenVisitor: SyntaxVisitor {
+  public var tree = [Node]()
+  public var current: Node!
+
+  var row = 0
+  var column = 0
+
+  override func visitPre(_ node: Syntax) {
+    var syntax = "\(type(of: node))"
+    if syntax.hasSuffix("Syntax") {
+      syntax = String(syntax.dropLast(6))
+    }
+
+    let node = Node(text: syntax)
+    node.range.startRow = row
+    node.range.startColumn = column
+    node.range.endRow = row
+    node.range.endColumn = column
+    if current == nil {
```

Comment:
when `current` is not an implicitly unwrapped optional, this could be rewritten with `if let current = current`

Reply:
this comment is still valid: you can avoid using optional current in the `else` clause by using `if let current = current`